velox/functions/prestosql/window/tests/NthValueTest.cpp

```diff
@@ -25,58 +25,100 @@ namespace {
 
 class NthValueTest : public WindowTestBase {
  protected:
+  NthValueTest() : frameClause_("range unbounded preceding") {}
+
+  NthValueTest(const std::string& frameClause) : frameClause_(frameClause) {}
+
+  void testWindowFunction(
+      const std::vector<RowVectorPtr>& input,
+      const std::string& function,
+      const std::vector<std::string>& overClauses) {
+    WindowTestBase::testWindowFunction(
+        input, function, overClauses, frameClause_);
+  }
+
   void testPrimitiveType(const TypePtr& type) {
     vector_size_t size = 25;
     auto vectors = makeRowVector({
         makeFlatVector<int32_t>(size, [](auto row) { return row % 5; }),
         makeFlatVector<int32_t>(size, [](auto row) { return row; }),
         makeFlatVector<int64_t>(size, [](auto row) { return row % 3 + 1; }),
-        typeValues(type, size),
+        makeFlatFuzzVector(type, size),
     });
-    testWindowFunction({vectors}, "nth_value(c3, c2)", kBasicOverClauses);
+
+    const std::vector<std::string> basicOverClauses = {
```

Comment:
The new clauses are added to get a deterministic ordering of rows, required for frames with k preceding / following bounds. The existing frame clauses do not provide a fixed ordering wrt column c3.

Reply:
Oh I see. A simpler way to specify that is by adding ", c3" as a suffix to the original kBasicOverClauses and kSortOrderBasedOverClauses instead of creating new lists. That will be easier to code as a loop. Please can you do so.

Reply:
Thanks for the suggestion, updated.